src/main/kotlin/no/nav/hjelpemidler/soknad/db/db/SøknadStore.kt

```diff
@@ -259,6 +265,29 @@ internal class SøknadStorePostgres(private val ds: DataSource) : SøknadStore {
                         ).asUpdate
                     )
                 }
+
+                runBlocking {
+                    launch(Job()) {
+
+                        // TODO consider extracting for new measurements
+                        val validStartStatuses = listOf(
+                            Status.GODKJENT,
+                            Status.GODKJENT_MED_FULLMAKT
+                        )
+                        val validEndStatuses = listOf(
+                            Status.VEDTAKSRESULTAT_ANNET,
+                            Status.VEDTAKSRESULTAT_AVSLÅTT,
+                            Status.VEDTAKSRESULTAT_DELVIS_INNVILGET,
+                            Status.VEDTAKSRESULTAT_INNVILGET,
+                            Status.VEDTAKSRESULTAT_MUNTLIG_INNVILGET
+                        )
+
+                        if (status in validEndStatuses)
+                            recordTimeElapsedBetweenStatusChange(session, soknadsId, SensuMetrics.TID_FRA_INNSENDT_TIL_VEDTAK, validStartStatuses, validEndStatuses)
```

Comment:
Er det behov for en try{} catch{} her?
Tenker dersom det skjer noe galt inne i `recordTimeElapsedBetweenStatusChange(...)`, kan det skape problem? At det ikke blir returnert noen verdi feks.

Reply:
har jeg brukt launch() på riktig måte? har prøvd at måling gå på sin egen coroutine slik at det ikke stopper hoved prosess hvis det oppstår feil. Tanke er det er ikke så kritisk hvis vi mister en datapunkt. Uansett tror det kunne gir oss noe info hvis vi logger en feil i catch blokk.

Reply:
Jeg vet ikke hvordan det fungerer med runBlocking. Med runBlocking vil det kjøre en async coroutine som om kodeblokken av synkron?

Reply:
jeg brukte noen timer i går for å undersøke og teste litt i noen scratch filer. med GlobalScope.launch() eller launch(Job()) kjøres rutine som utanfor konteksten til foreldren. På denne måte hovedprosessen venter ikke for at barnprocess er ferdig ;).
https://kotlinlang.org/docs/coroutine-context-and-dispatchers.html#children-of-a-coroutine

Reply:
Okay, men hva er funksjonen til runBlocking her?

Reply:
fra https://kotlinlang.org/docs/coroutines-basics.html#your-first-coroutine

runBlocking is also a coroutine builder that bridges the non-coroutine world of a regular fun main() and the code with coroutines inside of runBlocking { ... } curly braces. This is highlighted in an IDE by this: CoroutineScope hint right after the runBlocking opening curly brace.

If you remove or forget runBlocking in this code, you'll get an error on the launch call, since launch is declared only in the CoroutineScope: